overlord/devicestate/devicemgr.go

```diff
@@ -2940,6 +2952,78 @@ func (m *DeviceManager) checkEncryption(st *state.State, deviceCtx snapstate.Dev
 	return install.CheckEncryptionSupport(model, tpmMode, kernelInfo, gadgetInfo, m.runFDESetupHook)
 }
 
-func (m *DeviceManager) encryptionSupportInfo(model *asserts.Model, tpmMode secboot.TPMProvisionMode, kernelInfo *snap.Info, gadgetInfo *gadget.Info, systemSnapdVersions *install.SystemSnapdVersions) (install.EncryptionSupportInfo, error) {
-	return install.GetEncryptionSupportInfo(model, tpmMode, kernelInfo, gadgetInfo, systemSnapdVersions, m.runFDESetupHook)
+func (m *DeviceManager) encryptionSupportInfo(
+	systemAndSnaps *systemAndEssentialSnaps,
+	tpmMode secboot.TPMProvisionMode,
+	checkAction *secboot.PreinstallAction,
+	useCache bool,
```

Comment:
useCache is confusing term, because we use the cache sometimes even if it's false if we cached things.

Do we have cases where we don't save to the cache? it seems not

maybe dontRecomputeCache ?  anyway the main functions taking "useCache" flag need doc comment explaining what it does